typo corrections (#1564)
substrate-developer-hub/substrate-docs: content/md/en/docs/fundamentals/rust-basics.md

## Macros

> When you learn about writing FRAME pallets, you will quickly come across a variety of different types of macros designed to abstract and enforce any runtime specific requirements.

Reviewer: ```suggestion
As you learn how to use and write FRAME pallets, you'll see there are many macros available as reusable code to abstract common tasks or enforce runtime-specific requirements.
```

> With them, you're able to focus on writing idiomatic Rust, minimizing the overhead of writing extra code that you would otherwise need to write that correctly interacts with a runtime.

Reviewer: ```suggestion
These macros allow you to focus on writing idiomatic Rust and application-specific logic instead of the common code required to interact with a runtime.
```

> - Telling the compiler if code snippets are meant to compile to `no_std` or may access the `std` library or not.

Reviewer: ```suggestion
- Telling the compiler if code snippets are meant to compile to `no_std` or can access the `std` library.
```

> - Custom FRAME support macros for writing pallets.

Reviewer: ```suggestion
- Building custom FRAME pallets.
```
